source/guides/guides/test-retries.md

```diff
@@ -0,0 +1,210 @@
+
+```jsx
+{
+  "retries": {
+    // Configure retries for `cypress run`
+    // Default is 0
+    "runMode": 2,
+    // Configure retries for `cypress open`
+    // Default is 0
+    "openMode": 1
+  }
+}
+```
+
+## Custom Configurations
+
+### Individual Test(s)
+
+If you want a specific test to run a custom number of retries, you can set this by using the {% url "test's configuration" writing-and-organizing-tests#Test-Configuration %}.
+
+```jsx
+// Customize retries for an individual test
+describe('User sign-up and login', () => {
+  // `it` test block with no custom configuration
+  it('should redirect unauthenticated user to sign-in page', () => {
+    // ...
+  })
+
+  // `it` test block with custom configuration
+  it('allows user to login', {
+    retries: {
+      runMode: 2,
+      openMode: 1
+    }
+  }, () => {
+    // ...
+  })
+})
+```
+
+### Test Suite(s)
+
+If you want a suite of tests to re-run a custom number of times, you can do this by setting the suite's configuration.
+
+```jsx
+// Customizing retries for a suite of tests
+describe('User bank accounts', {
+  retries: {
+    runMode: 2,
+    openMode: 1,
+  }
+}, () => {
+  // Individual tests will be run normally
+  it('allows a user to view their transactions, () => {
+    // ...
+  }
+
+  it('allows a user to edit their transactions, () => {
+    // ...
+  }
+})
+```
+
+You can find more information about custom configurations here: {% url "Test Configuration" configuration#Test-Configuration %}
+
+# Screenshots
+
+With test retries, Cypress will take a separate screenshot for each test attempt. These will be properly labeled with the attempt number appended at the end of each filename.
+
+With the following test code, you would see the below screenshot filename when all 3 attempts fail:
+
+```js
+describe('User Login', () => {
+  it('displays login errors', () => {
+    cy.visit('/')
+    cy.screenshot()
+    // ...
+  })
+})
+```
+
+```js
+// screenshot filename from cy.screenshot() on 1st attempt
+'User Login -- displays login errors.png'
+// screenshot filename on 1st failed attempt
+'User Login -- displays login errors (failed).png'
+// screenshot filename from cy.screenshot() on 2nd attempt
+'User Login -- displays login errors (attempt 2).png'
+// screenshot filename on 2nd failed attempt
+'User Login -- displays login errors (failed) (attempt 2).png'
+// screenshot filename from cy.screenshot() on 3rd attempt
+'User Login -- displays login errors (attempt 3).png'
+// screenshot filename on 3rd failed attempt
+'User Login -- displays login errors (failed) (attempt 3).png'
+```
+
+# Dashboard
+
+If you are using the {% url "Cypress Dashboard" dashboard %}, test retries will not change your current experience at this time. However, if you are interested in future enhancements coming to the Dashboard experience, be sure to check out our {% url "product roadmap" https://cypress-io.productboard.com/roadmap/1238172-product-roadmap %}.
```

Comment:
```suggestion
If you are using the {% url "Cypress Dashboard" dashboard %}, information related to test retries is not currently shown. Surfacing this information and other analytics related to test retries is on our {% url "product roadmap" https://cypress-io.productboard.com/roadmap/1238172-product-roadmap %}.
```

Reply:
Well, it will show the extra screenshots though and show the retries in the recorded video, so this isn't entirely accurate right? Hmm...